File: recursion/factorial.py
# factorial function
def factorial(n):
  # break condition for recursion
  if n==1 or n==0:
    return 1
  else:
    return n*factorial(n-1)

num = int(input())
fact = factorial(num)
print('Factorial of {0} is {1}'.format(num,fact))

File: recursion/factorial.cpp
#include<iostream>
using namespace std;
int fact(int num)
{
  if(num==0)
    return 1;
  else
    return num*fact(num-1);
}
int main()
{
	int num;
	cin<<num;
	cout<<fact(num);
}
      
